src/components/NavBar/NavBar.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { Children, ReactNode, useRef, useEffect } from 'react';
+import { Link, NavLink, useLocation } from 'react-router-dom';
+import { ReactComponent as IconClose } from 'components/Modal/x-mark.svg';
+import { ReactComponent as IconMenu } from './Hamburger.svg';
+import { ReactComponent as IconHome } from './Logo.svg';
+import { ReactComponent as IconModules } from './Modules.svg';
+import { ReactComponent as IconNotebook } from './Notebook.svg';
+
+import styles from './NavBar.module.css';
+
+type NavBarProps = {
+  children?: ReactNode;
+};
+
+export default function NavBar({ children }: NavBarProps) {
+  const { pathname } = useLocation();
+
+  const menuRef = useRef<HTMLUListElement>(null);
+
+  const toogleMobileMenu = () => {
+    menuRef.current?.classList.toggle('active');
+  };
+
+  useEffect(() => {
+    const focusableElements = menuRef.current?.querySelectorAll('a, button');
+    const firstElement =
+      focusableElements && (focusableElements[0] as HTMLElement);
+    const lastElement =
+      focusableElements &&
+      (focusableElements[focusableElements.length - 1] as HTMLElement);
+
+    const focusTrap = (e: KeyboardEvent) => {
```

Comment:
I will definitely study this so that I can also work on accessibility in our project